install/first_google_boot.sh

```diff
@@ -215,21 +219,42 @@ function extract_spinnaker_kube_credentials() {
     kube_zone=$MY_ZONE
   fi
 
+  if clear_metadata_to_file "kube_config" $config_path; then
```

Comment:
This function still feels complicated the way it flows. It appears this function is responsible for determining if kubernetes should be enabled or not and, if so, configuring it. This involves three things:
  0) determining if kubernetes should be enabled or not
if so then:
  1) writing the credentials to .kube/config
  2) writing the SPINNAKER_KUBERNETES_ENABLED flag to /etc/default/spinnaker

(0) is inferred by having credentials (which happens in (1))
(1) either gets explicit credentials from "kube_config" or creates its own credentials by interacting with the cluster specified by "kube_(project|cluster|zone)".

So the flow should read more like
   if (have_kube_credentials) {
        write ENABLED
   }
where have_kube_credentials is a helper function that:
    if (have "kube_config") {
        write config_path
        clear kube_(cluster|zone|project) since we dont need them
        return true
    }
   if can generate credentials from others
        return true
   return false

You can unroll this without the helper function but the flow is similar, especially isolating the two different ways to produce credentials from everything else, which is the same regardless of how you got the credentials.

Especially since you dont use the other parameters. If you were configuring the actual kubernetes cluster, zone, project even if you had kube_config then the flow might look different. So I think it's that much more important to clarify.